src/v/cluster/rm_stm.cc

```diff
@@ -1502,6 +1502,12 @@ ss::future<bool> rm_stm::sync(model::timeout_clock::duration timeout) {
     auto ready = co_await persisted_stm::sync(timeout);
     if (ready) {
         if (_mem_state.term != _insync_term) {
+            // There is a term change, issue a checkpoint purge to followers
+            // All other replicate ops need to wait until this finishes, this is
+            // so that the followers clean up the state before we start applying
+            // changes to the in memory state.
+            co_await _checkpoint_applied_replicate.with(
+              [this]() { return replicate_checkpoint_applied(_insync_term); });
```

Comment:
Should we need to use new feature here? 
You introduced new record type. So you should protect upgrade process

Reply:
good point. i wonder if old nodes ignore unknown comamnds? probably not